openedx/core/lib/api/paginators.py

```diff
@@ -15,6 +15,7 @@ class DefaultPagination(pagination.PageNumberPagination):
     by any subclass of Django Rest Framework's generic API views.
     """
     page_size_query_param = "page_size"
+    max_page_size = 100
```

Comment:
@nasthagiri The default page size did not change. I put a cap on the maximum size page a client can request. This is plugging a hole. We have 'page_size_query_param' set, but not 'max_page_size'. The REST framework documentation states that if you want to give a client the ability to specify their page size, you must specify a parameter name with 'page_size_query_param'. You then are to specify the max size that should be allowed. We never did that, so, in theory, one can pull arbitrarily large pages. The paginators we use handle ten items at a time, but I figured there might be occasions I missed where someone used more. 100 seemed like a reasonable cap.